fl4health/reporting/base_reporter.py

```diff
@@ -0,0 +1,51 @@
+"""Base Class for Reporters.
+
+Super simple for now but keeping it in a seperate file in case we ass more base methods.
+"""
+
+from typing import Any
+
+
+class BaseReporter:
+    def report(
+        self,
+        data: dict,
+        round: int | None = None,
+        epoch: int | None = None,
+        batch: int | None = None,
+    ) -> None:
+        """A method called by clients or servers to send data to the reporter.
+
+        The report method is called reduncantly by the clients every step, every epoch,
```

Comment:
nit: reduncantly -> redundantly

Reply:
The report method is called by the client/server at frequent intervals (ie step, epoch, round) and sometimes outside of a FL round (for high level summary data). It is the responsibility of the Reporter to determine when and what to report.

Reply:
Just to not only be specific to client